exp/internal/controllers/machinepool_controller.go

```diff
@@ -377,6 +379,7 @@ func (r *MachinePoolReconciler) watchClusterNodes(ctx context.Context, cluster *
 		Watcher:      r.controller,
 		Kind:         &corev1.Node{},
 		EventHandler: handler.EnqueueRequestsFromMapFunc(r.nodeToMachinePool),
+		Predicates:   []predicate.TypedPredicate[client.Object]{predicates.TypedResourceIsChanged[client.Object](log)},
```

Comment:
We can't use this log. It contains all k/v pairs of the current reconcile

(same problem as here: https://github.com/kubernetes-sigs/cluster-api/pull/11406)

Let's store and use the predicateLog in the reconciler